packages/graphql/src/modules/user/app/authenticateFacebookUser.spec.ts

```diff
@@ -0,0 +1,197 @@
+import User from '../domain/User';
+import createFakeStorageProvider from '../../../shared/providers/StorageProvider/fakes/fakeStorageProvider';
+import createFakeUsersRepository from '../infra/fakes/fakeUsersRepository';
+import createFakeSocialMediaRepository from '../infra/fakes/fakeSocialMediaRepository';
+import createFakeTokenProvider from '../providers/TokenProvider/fakes/fakeTokenProvider';
+
+import authenticateFacebookUser from './authenticateFacebookUser';
+
+const authenticateUserParams = {
+  token: 'fake-token',
+  permissions: ['public_profile', 'email'],
+  expires: 60 * 60 * 24 * 30,
+};
+
+const fakeUser = new User({
+  id: 'fake-uuid',
+  name: 'fake-user',
+  emails: [
+    {
+      email: 'mysocialemail@fake.com',
+      providedBy: 'facebook',
+    },
+  ],
+  profileImages: [
+    {
+      providedBy: 'facebook',
+      url: 'fake-token/profilePicture.jpeg',
+    },
+  ],
+  providersIds: [
+    {
+      id: 'socialId-fake-token',
+      providedBy: 'facebook',
+    },
+  ],
+});
+
+const setUp = () => {
+  const userRepository = createFakeUsersRepository();
+  const storageProvider = createFakeStorageProvider();
+  const facebookRepository = createFakeSocialMediaRepository();
+  const tokenProvider = createFakeTokenProvider({userRepository});
+  const authenticateUser = authenticateFacebookUser({
+    facebookRepository,
+    storageProvider,
+    tokenProvider,
+    userRepository,
+  });
+
+  return {
+    authenticateUser,
+    storageProvider,
+    userRepository,
+    facebookRepository,
+    tokenProvider,
+  };
+};
+
+describe('AuthenticateFacebookUser', () => {
+  it('should be able to authenticate', async () => {
+    const {authenticateUser} = setUp();
+
+    const {user, token} = await authenticateUser(authenticateUserParams);
+
+    expect(user).toBeInstanceOf(User);
+    expect(user).toEqual({
+      id: 'fake-uuid',
+      name: 'fake-user',
+      emails: [
+        {
+          email: 'mysocialemail@fake.com',
+          providedBy: 'facebook',
+        },
+      ],
+      profileImages: [
+        {
+          providedBy: 'facebook',
+          url: 'fake-token/profilePicture.jpeg',
+        },
+      ],
+      providersIds: [
+        {
+          id: 'socialId-fake-token',
+          providedBy: 'facebook',
+        },
+      ],
+    });
+    expect(token).toMatch(/(JWT )\w+\.\w+\.\w+/g);
+  });
+
+  it('should be able to authenticate with no user photo', async () => {
```

Comment:
Can we break down these different contexts in describe blocks?
Something like:
```js
describe('When the user has no photo', () => {
  it('authenticates it, () => {
   ....
  });
})

Reply:
The same for all others below

Reply:
That's a good suggestion, thanks!

Reply:
Okay 👍🏼